lib/pure/math.nim

```diff
@@ -941,6 +941,52 @@ func euclMod*[T: SomeNumber](x, y: T): T {.since: (1, 5, 1).} =
   if result < 0:
     result += abs(y)
 
+func ceilDiv*[T: SomeInteger](x, y: T): T {.inline, since: (1, 5, 1).} =
+  ## Ceil division is conceptually defined as `ceil(x / y)`.
+  ##
+  ## Assumes `x >= 0` and `y > 0` and (`x + y - 1 <= high(T)` if T is SomeUnsignedInt).
+  ##
+  ## This is different from the `system.div <system.html#div,int,int>`_
+  ## operator, which is defined as `trunc(x / y)`.
+  ## That is, `div` rounds towards `0` and `ceilDiv` rounds up.
+  ##
+  ## This function has above input limitation because it is implemented with
+  ## the expression that backend compiler generates faster code and it is
+  ## rarely used with negative value or unsigned int close to high(T)/2.
+  ## If you need `ceilDiv` works with any input, see:
+  ## https://github.com/demotomohiro/divmath
+  ##
+  ## **See also:**
+  ## * `system.div proc <system.html#div,int,int>`_ for integer division
+  ## * `floorDiv func <#floorDiv,T,T>`_ for integer division with rounds down.
+  runnableExamples:
+    assert ceilDiv( 12,  3) ==  4
+    assert ceilDiv( 13,  3) ==  5
+
+  when sizeof(T) == 8:
+    type UT = uint64
+  elif sizeof(T) == 4:
+    type UT = uint32
+  elif sizeof(T) == 2:
+    type UT = uint16
+  else:
+    type UT = uint8
```

Comment:
Is it theoretically possible that `sizeof(T)` is 16 or something? And do we really want to use `uint32`/`uint64` if `T` is `int` (not sure if this would make a difference, but #18445 seems like a better solution).

Reply:
As T is SomeInteger, T can be int8 or int16.
If arguments were signed int and they were not converted to unsigned int,
it doesn't work any positive signed int value because `x + (y - 1)` can overflow.
`toUnsigned` in https://github.com/nim-lang/Nim/pull/18445 would be better,
but I cannot use it until it is mergerd.

Reply:
I don't see how this is relevant to my question... I'm asking if it is possible that `T` (in particular `int`, since the others have a fixed size) can be 16 bytes (i.e. 128 bits) or more. In that case, `UT` would be `uint8`, which would be wrong.

Reply:
I'm sorry for misunderstanding your question.
I don't think sizeof(T) can be 16 or larger as there is no `int128` or `uint128` types in `system` module.
https://nim-lang.github.io/Nim/system.html
But I fixed the code so that when `ceilDiv` is called with int type such like sizeof(T) > 8, it becomes compile error.